### DISCUSSION TITLE
commands of "brew services" failed after upgrading to Mac OS 11 Big Sur

### DISCUSSION THREAD

**Discussion by zoranhasan** [Homebrew/discussions#451] [Everyday usage] [answered]
After I upgraded my system to Mac OS 11 Big Sur, I met a problem whenever I ran a command of "brew services" including "brew services list" or "brew services start xxx".
The error messages are displayed as:
```
Error: undefined method `args' for Homebrew:Module
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-services/lib/services_cli.rb:65:in `run!'
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-services/cmd/services.rb:48:in `services'
/usr/local/Homebrew/Library/Homebrew/brew.rb:124:in `<main>'
```

I have googled such problem but got no relevant results, is there any similar problem which any one ever met?

Thanks.

**Comment by SMillerDev** [comment, marked as answer]
Try running `brew update` and `brew upgrade`. If the problem persists, post the output of `brew doctor`

**Reply by zoranhasan** [reply]
Thanks, it worked after 'brew update' command finished.

**Reply by tom10271** [reply]
Run `brew upgrade` at your own risk, if softwares you installed are deprecated, it is likely some of your applications will be broken
